api/queries_pr.go

```diff
@@ -576,9 +577,13 @@ loop:
 		}
 
 		for _, edge := range prData.Edges {
-			prs = append(prs, edge.Node)
-			if len(prs) == limit {
-				break loop
+			if _, ok := check[edge.Node.Number]; !ok {
```

Comment:
Nit: to avoid deeper nesting in go, could we flip this condition and have it `continue` instead?

```go
if _, exists := check[edge.Node.Number]; exists {
  continue
}
```